mne_bids/io.py

```diff
@@ -105,9 +105,22 @@ def read_raw_bids(bids_fname, bids_root, return_events=True,
     bids_root : str
         Path to root of the BIDS folder
     return_events: bool
-        Weather to return events or not. Default is True.
+        Whether to return events (and event_id) or not.
+        Default is True.
     verbose : bool
         The verbosity level
+
+    Returns
+    -------
+    raw : instance of Raw
+        The MNE raw object containing the data.
+    events : array, (n_events, 3)
```

Comment:
Should there be an `, optional` for the two parameters that are only returned if `return_events` is `True`?
Also this raises a question... Is it good to have a function that can return differing numbers of objects?

Reply:
I've seen [this kind of code in MNE](https://github.com/mne-tools/mne-python/blob/73ea718b82c7ea5f09b70490ca3ba7de2fd79f78/mne/time_frequency/multitaper.py#L139). I'm not sure if it's a good idea in public API but I will defer to the opinion of @agramfort :)

Reply:
I haven't seen any indication that an `, optional` should be used for multiple return values (some of which are optional).

That apart, I do think it's a confusing practice, see also [this discussion](https://grokbase.com/t/python/python-list/028wpwxmvt/stylistic-question-optional-return-value/oldest)

I think it'd be more helpful if the same amount of outputs is always returned ...